benchmarks/performancepoetry/src/main/java/com/squareup/benchmarks/performance/poetry/PerformancePoemWorkflow.kt

```diff
@@ -0,0 +1,235 @@
+import kotlinx.coroutines.flow.MutableStateFlow
+
+/**
+ * Version of [PoemWorkflow] that takes in a [SimulatedPerfConfig] to control the performance
+ * behavior of the Workflow.
+ *
+ * @param [simulatedPerfConfig] specifies whether to make the Workflow more 'complex' by
+ * introducing some asynchronous delays. See [SimulatedPerfConfig] for more details.
+ *
+ * @param [isLoading] will be set to true while this workflow is 'loading'. This is so that another
+ * component, such as a [MaybeLoadingGatekeeperWorkflow] can overlay the screen with a visual
+ * loading state. N.B. that whether or not this is loading could be included in the
+ * RenderingT if the interface [PoemWorkflow] had been left more flexible.
+ *
+ * ** Also note that raw mutable state sharing like this will almost always be a smell. It would
+ * be better to inject an interface of a 'Loading' service that could trigger this and likely
+ * break ties/conflicts with a token in the start/stop requests. We leave that complexity out
+ * here. **
+ */
+class PerformancePoemWorkflow(
+  private val simulatedPerfConfig: SimulatedPerfConfig = SimulatedPerfConfig.NO_SIMULATED_PERF,
+  private val isLoading: MutableStateFlow<Boolean>
+) : PoemWorkflow, StatefulWorkflow<Poem, State, ClosePoem, OverviewDetailScreen>() {
+
+  sealed class State {
+    // N.B. This state is a smell. We include it to be able to mimic smells
+    // we encounter in real life. Best practice would be to fold it
+    // into [Selected(NO_SELECTED_STANZA)] at the very least.
+    object Initializing : State()
+    data class ComplexCall(
+      val payload: Int
+    ) : State()
+
+    data class Selected(val stanzaIndex: Int) : State()
+  }
+
+  override fun initialState(
+    props: Poem,
+    snapshot: Snapshot?
+  ): State {
+    return if (simulatedPerfConfig.useInitializingState) Initializing else Selected(
+      NO_SELECTED_STANZA
+    )
+  }
+
+  @OptIn(WorkflowUiExperimentalApi::class)
+  override fun render(
+    renderProps: Poem,
+    renderState: State,
+    context: RenderContext
+  ): OverviewDetailScreen {
+    return when (renderState) {
+      Initializing -> {
+        context.runningWorker(Worker.from { Unit }, "initializing") {
+        isLoading.value = true
+        action {
+          isLoading.value = false
+          state = Selected(NO_SELECTED_STANZA)
+        }
+      }
+        OverviewDetailScreen(overviewRendering = BackStackScreen(BlankScreen))
+      }
+      else -> {
+        val (stanzaIndex, currentStateIsLoading) = when (renderState) {
+          is ComplexCall -> Pair(renderState.payload, true)
+          is Selected -> Pair(renderState.stanzaIndex, false)
+          Initializing -> throw IllegalStateException("No longer initializing.")
+        }
+
+        if (currentStateIsLoading) {
+          context.runningWorker(
+            Worker.from {
+              isLoading.value = true
+              delay(simulatedPerfConfig.complexityDelay)
+              stanzaIndex
```

Comment:
Seems like you're not actually using the `stanzaIndex` value in the action below?